modules/gapi/src/streaming/onevpl/engine/preproc/preproc_engine.cpp

```diff
@@ -0,0 +1,501 @@
+                do {
+                    if (!my_sess.procesing_surface_ptr.expired()) {
+                        session_type::op_handle_t pending_op = my_sess.sync_in_queue.front();
+                        GAPI_LOG_DEBUG(nullptr, "pending IN operations count: " <<
+                                                my_sess.sync_in_queue.size() <<
+                                                ", sync id:  " <<
+                                                pending_op.first <<
+                                                ", surface:  " <<
+                                                pending_op.second);
+
+                        my_sess.sync_in_queue.pop();
+                        auto *dec_surface = pending_op.second;
+                        auto *vpp_suface = my_sess.procesing_surface_ptr.lock()->get_handle();
+
+                       /* TODO: consider CROP/ROI here
+                        static int x_offset = 0;
+                        static int y_offset = 0;
+                        dec_surface->Info.CropX = x_offset;
+                        dec_surface->Info.CropY = y_offset;
+                        dec_surface->Info.CropW = 100 + x_offset++;
+                        dec_surface->Info.CropH = 100 + y_offset++;
+                        */
+                        my_sess.last_status = MFXVideoVPP_RunFrameVPPAsync(my_sess.session,
+                                                                           dec_surface,
+                                                                           vpp_suface,
+                                                                           nullptr, &pending_op.first);
+                        pending_op.second = vpp_suface;
+
+                        GAPI_LOG_DEBUG(nullptr, "Got VPP async operation" <<
+                                                ", sync id:  " <<
+                                                pending_op.first <<
+                                                ", dec surface:  " <<
+                                                dec_surface <<
+                                                ", trans surface: " <<
+                                                pending_op.second <<
+                                                ", status: " <<
+                                                mfxstatus_to_string(my_sess.last_status));
+
+                        // NB: independently from result code
+                        // we no need in decoding frame anymore: because we had pulled
+                        // decoding frames to this point by onevpl::Surface highlevel locking
+                        // mechanism and after that we adhere existing MFX/VPL inner locking mechanism
+                        // on native mfxSurface pointer
+                        abandon_decode_frame(dec_surface);
+
+                        GAPI_Assert(dec_surface != pending_op.second &&
+                                    "Unexpected out VPP surface");
```

Comment:
no i meant that `MFXVideoVPP_RunFrameVPPAsync`  must return `vpp surface` and it must be not the same as `decode surface`
It just sanity check to figure out how VPP works, it may be removed completely now.

Will delete it